### PR TITLE
Add HEADER.md file

### PR DESCRIPTION
**Pull request by harshavamsi**
Signed-off-by: Harsha Vamsi Kalluri <harshavamsi096@gmail.com>

### Description
Adds recommended license headers for every language.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: HEADERS.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+```
+
+If files have any existing headers keep them and add OpenSearch SPDX license header on top.  
+_Example_: [Version.java](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/Version.java) file has license header from Elasticsearch.
+ The existing license is not removed and OpenSearch SPDX license header is added on top of it.   
+
+### Automated License Headers' Checks
```
Comment (dblock) [inline HEADERS.md:56]: This is talking about plugins. I think it belongs in opensearch-plugins.

File: HEADERS.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- [Headers](#headers)
+  - [License Headers](#license-headers)
+    - [Java, Javascript, .NET, PHP, Rust](#java-javascript-net-php-rust)
+    - [Python, Ruby, Shell](#python-ruby-shell)
+    - [Go](#go)
+  - [Automated License Headers' Checks](#automated-license-headers-checks)
+<!-- TOC -->
+## Headers
+
+### License Headers
```
Comment (dblock) [inline HEADERS.md:11]: This is generic for all repos in this project.

**Review by dblock — CHANGES_REQUESTED** [review]
I’d like most of this content to go into .github and only client-specific things to be here. You can link back to that repo from wherever appropriate.

**Comment by harshavamsi** [comment]
@dblock I've removed most of the content and linked back to .github's license text.

**Comment by dblock** [comment]
So there's nothing client-specific anymore :) Do we need this file at all?

**Comment by harshavamsi** [comment]
> So there's nothing client-specific anymore :) Do we need this file at all?

I don't think so :) Closing.

**Pull request closed without merging**